File: GenerateGBFSDataset/src/main/scala/com/david/GenerateGBFS/httpclient/HttpClientBasicFactory.scala
package com.david.GenerateGBFS.httpclient

import org.apache.http.impl.client.{CloseableHttpClient, HttpClients}


object HttpClientBasicFactory extends IHttpClientBasicFactory with AutoCloseable{
  private lazy val httpClient: CloseableHttpClient = HttpClients.createDefault()
  override lazy val getHttpClient: CloseableHttpClient = httpClient
  override lazy val close: Unit = httpClient.close()
}

File: GBFSCron/src/main/java/com/david/GBFSCron/utils/httpclient/HttpClientBasicFactory.java
package com.david.GBFSCron.utils.httpclient;

import org.apache.http.impl.client.CloseableHttpClient;
import org.apache.http.impl.client.HttpClients;

import java.io.Closeable;
import java.io.IOException;

public class HttpClientBasicFactory implements IHttpClientFactory, Closeable {
    private CloseableHttpClient httpClient;

    HttpClientBasicFactory(){
        httpClient = HttpClients.createDefault();
    }

    @Override
    public void close() throws IOException {
        httpClient.close();

    }

    @Override
    public CloseableHttpClient getHttpClient() {
        return httpClient;
    }

}